src/thor/bidirectional_astar.cc

```diff
@@ -982,5 +993,111 @@ std::vector<std::vector<PathInfo>> BidirectionalAStar::FormPath(GraphReader& gra
   return paths;
 }
 
+bool IsBridgingEdgeRestricted(GraphReader& graphreader,
+                              std::vector<sif::BDEdgeLabel>& edge_labels_fwd,
+                              std::vector<sif::BDEdgeLabel>& edge_labels_rev,
+                              const BDEdgeLabel& fwd_pred,
+                              const BDEdgeLabel& rev_pred,
+                              std::shared_ptr<sif::DynamicCost>& costing) {
+
+  const uint8_t M = 10;                 // TODO Look at data to figure this out
+  const uint8_t PATCH_PATH_SIZE = M * 2 // Expand M in both directions
+                                  + 1;  // Also need space for pred in the middle
+
+  // Begin by building the "patch" path
+  std::vector<GraphId> patch_path;
+  patch_path.reserve(PATCH_PATH_SIZE);
+
+  patch_path.push_back(fwd_pred.edgeid());
+
+  auto next_fwd_pred = fwd_pred;
+  for (int i = 0; i < M; ++i) {
+    // Walk M edges back and add each to patch path
+    const uint32_t next_pred_idx = next_fwd_pred.predecessor();
+    if (next_pred_idx == baldr::kInvalidLabel) {
+      break;
+    }
+    next_fwd_pred = edge_labels_fwd[next_pred_idx];
+    if (!next_fwd_pred.on_complex_rest()) {
+      // We can actually stop here if this edge is no longer path of any complex restriction
+      break;
+    }
+
+    patch_path.push_back(next_fwd_pred.edgeid());
+  }
+
+  // Reverse patch_path so that the leftmost edge is first and original `pred`
+  // at the end before pushing the right-hand edges (opposite direction) onto the back
+  std::reverse(patch_path.begin(), patch_path.end());
+
+  const GraphTile* tile = nullptr; // Used for later hinting
+
+  auto next_rev_pred = rev_pred;
+  // Now push_back the edges from opposite direction onto our patch_path
+  for (int n = 0; n < PATCH_PATH_SIZE; ++n) {
+    auto next_rev_pred_idx = next_rev_pred.predecessor();
+    if (next_rev_pred_idx == kInvalidLabel) {
+      // We reached the end of the opposing tree, i.e. destination or origin
+      break;
+    }
+    next_rev_pred = edge_labels_rev[next_rev_pred_idx];
+    if (!next_rev_pred.on_complex_rest()) {
+      // We can actually stop here if this edge is no longer path of any complex restriction
+      break;
+    }
+
+    // Since we are on the reverse expansion here, we want the opp_edgeid
+    // since we're tracking everything in the forward direction
+    const auto edgeid = next_rev_pred.opp_edgeid();
+    patch_path.push_back(edgeid);
+
+    // Also grab restrictions while walking for later comparison against patch_path
+    //
+    tile = graphreader.GetGraphTile(edgeid, tile);
+    if (tile == nullptr) {
+      throw std::logic_error("Tile pointer was null in IsBridgingEdgeRestricted");
```

Comment:
this just cant happen, because the edge is coming from the path that the algorithm was expanding

Reply:
I think it is fine to have some kind of check around "impossible" states. Would you prefer to log an error message or some other effect?